website/addons/osfstorage/views.py

```diff
@@ -164,14 +164,17 @@ def osfstorage_create_child(file_node, payload, node_addon, **kwargs):
                     })
                 ),
                 dict(payload['metadata'], **payload['hashes'])
-            )
+            ).find_matching_archive()
```

Comment:
1. Would it not be more appropriate for the `find_matching_archive` to be part of the OsfStorageFileVersion constructor?

2. Seems `find_matching_archive` should be a private method used on construction, removing the need for the `save` input.

3. We should return the version from `create_version` and return it rather than the skimming the last object below `file_node.versions[-1]`.

4. Archive could now be imputed directly by the vault's field existing in the version record.